CDS/scripts/js_utils/youtube/youtube-upload-lib.js

```diff
@@ -65,37 +65,58 @@ function getYoutubeData(connection) {
     });
 }
 
+function addPosterImage(connection, videoId, youtubeClient, account) {
+    return dataStore.get(connection, 'meta', 'poster_image')
+    .then(file => {
+
+        if (file.value) {
+
+          if (!account) {
+            return new Promise((fulfill, reject) => { reject( new Error('could not add a poster image: missing account owner') )});
+          }
+          return  new Promise((fulfill, reject) => {
+            youtubeClient.thumbnails.set({ videoId: videoId, onBehalfOfContentOwner: account, media: {body: fs.createReadStream(file.value)}}, (err, result) => {
+                  if (err) reject(err);
+                  else fulfill(result);
+            });
+          });
+        }
+        return new Promise(fulfill => {fulfill()});
```

Comment:
apologies if I'm being dense, but what effect would this have if poster_image was not set in the datastore? an empty result would then be fulfilled I guess.  would it not be more technically accurate to reject?